### PR TITLE
New Version: Introduction to Ray

### PR DESCRIPTION
**Pull request by emmyscode**
This updated version of the "Introduction to Ray" with a few major differences:
- removed quick sort example
- added simple intro example about remote functions
- added method tracking example with Ray Actors
- updated image assets
- general tweaking of language
- update the README on the landing page for the repo

### CODE REVIEW

File: Introductory_modules/Introduction_to_Ray.ipynb [hunk 1]
```diff
@@ -69,26 +68,371 @@
...
+    "\n",
+    "print(duration)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 2. Remote Objects Example\n",
+    "\n",
```
Comment (dmatrix) [inline Introductory_modules/Introduction_to_Ray.ipynb:233]: May be couple of sentence what object store is in Ray. for example,

"Objects are immutable and can be accessed from anywhere on the cluster, as they are stored in the shared memory. An object ref is essentially a pointer or a unique ID that can refer to a remote object without seeing its value. If you’re familiar with futures in Python, Java, or Scala, Ray object refs are conceptually similar."

File: Introductory_modules/Introduction_to_Ray.ipynb [hunk 2]
```diff
@@ -69,26 +68,371 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 4. Housing Prices with `sklearn`\n",
     "***\n",
-    "Let's take a look at how we can apply Ray Core's flexible and simple API to scale a bare bones version of a common ML task: cross-validation.\n",
+    "Now that we've warmed-up with some isolated examples using tasks and actors, let's take a look at how we can apply Ray Core's flexible and simple API to scale a bare bones version of a common ML task: cross-validation.\n",
```
Comment (dmatrix) [inline Introductory_modules/Introduction_to_Ray.ipynb:435]: Now that we've warmed-up with some isolated examples using tasks, objects, and actors, let's ....

File: Introductory_modules/Introduction_to_Ray.ipynb [hunk 3]
```diff
@@ -411,6 +636,8 @@
     "# Homework\n",
     "---\n",
     "If you would like to practice your new skills further with some in-depth examples beyond the embedded coding excercises, take a look at this list of suggested problems:\n",
+    "- [Read About Debugging and Profiling on Ray](https://docs.ray.io/en/latest/ray-core/troubleshooting.html)\n",
+    "    - Dig into how to observe Ray work by visualizing tasks in the Ray timeline, profiling using Python's CProfile, understanding crashes and suboptimal performance, and more in this user guide.\n",
     "- [Distribute a Classical Algorithm with Ray](https://github.com/ray-project/hackathon5-algo)\n",
     "    - In this excercise, go to the GitHub repo linked above for details on choosing a classic algorithm implemented in Python, editing the implementation to parallelize the work with Ray, and compare your results against the sequential implementation.\n",
     "\n",
```
Comment (dmatrix) [inline Introductory_modules/Introduction_to_Ray.ipynb:643]: Might want add reference to further use of Ray core examples: https://docs.ray.io/en/latest/ray-core/examples/overview.html

**Review by dmatrix — APPROVED** [review]
Added minor comments. Good quick start intro to Ray Core concepts.

File: Introductory_modules/Introduction_to_Ray.ipynb [hunk 4]
```diff
@@ -69,26 +68,371 @@
...
+    "  return x\n",
+    "```\n",
+    "This is a Ray remote function. To run it, you would append the postfix `.remote()` onto this function call, i.e. `quick_nap.remote(2)`. Instead of returning a `2` after 1 second like the previous function, this function will immediately return an `ObjectRef`. For example:\n",
+    "\n",
+    "```\n",
+    "ObjectRef(e0dc174c83599034ffffffffffffffffffffffff0100000001000000)\n",
+    "```\n",
+    "This `ObjectRef` is a *promise* of future work, meaning that the actual task of the function is delegated in the background to a worker. In order to access the expected output, you need to call `ray.get()` on the `ObjectRef`. Try it out in the coding excercise below!\n"
    ]
   },
```
[the remark this answers is not in the document]
Comment (emmyscode) [reply Introductory_modules/Introduction_to_Ray.ipynb:110]: Done!

File: Introductory_modules/Introduction_to_Ray.ipynb (hunk 1 above)
Comment (emmyscode) [reply Introductory_modules/Introduction_to_Ray.ipynb:233]: Done!

File: Introductory_modules/Introduction_to_Ray.ipynb (hunk 2 above)
Comment (emmyscode) [reply Introductory_modules/Introduction_to_Ray.ipynb]: Done!

File: Introductory_modules/Introduction_to_Ray.ipynb (hunk 3 above)
Comment (emmyscode) [reply Introductory_modules/Introduction_to_Ray.ipynb:648]: Done!

**Pull request merged**